Decode/PerfItemType.cs

```diff
@@ -103,37 +164,52 @@ namespace Microsoft.LinuxTracepoints.Decode
         /// <summary>
         /// For simple encodings (e.g. Value8, Value16, Value32, Value64, Value128),
         /// this is the size of one element in bytes (1, 2, 4, 8, 16). For complex types
-        /// (e.g. Struct or string), this is 0.
+        /// (e.g. Struct or String), this is 0.
         /// </summary>
         public byte TypeSize { get; }
 
+        /// <summary>
+        /// Returns Encoding | ArrayFlag | (IsScalar ? ChainFlag : 0).
+        /// </summary>
+        private EventHeaderFieldEncoding EncodingAndArrayFlagAndIsScalar { get; }
```

Comment:
Note: Rename property `EncodingAndArrayFlags` to `EncodingAndArrayFlagAndIsScalar`, and make it private. Storage of these three properties is an implementation detail that doesn't need to be exposed.